Add Sign.recoverV()

mialbu opened the pull request:
Resolves #949

gsmachado reviewed (dismissed):
hmmm, LGTM. 👍 
🚀 🥇

gsmachado commented:
Let's wait for @csmuller to give his opinion.

csmuller commented:
@mialbu Apologies for directly fiddling with your code.... I added a second test with `v = 28` and added an overload of the recover method where you can use `SignatureData` as the sig param. This is because it is not always intuitive for the user to create two `BigInteger`s. In my test case I had to use the `BigInteger(int signum, byte[] magnitude)` constructor because otherwise `r` would have been interpreted as a negative number.

mialbu commented:
> @mialbu Apologies for directly fiddling with your code.... I added a second test with `v = 28` and added an overload of the recover method where you can use `SignatureData` as the sig param. This is because it is not always intuitive for the user to create two `BigInteger`s. In my test case I had to use the `BigInteger(int signum, byte[] magnitude)` constructor because otherwise `r` would have been interpreted as a negative number.

Great additions!💯

gsmachado commented:
BTW: there's something Codacy is complaining:

https://app.codacy.com/gh/neow3j/neow3j/pullRequest?prid=12371440

Maybe consider changing that to increase code quality. 😄

csmuller commented on core/src/main/java/io/neow3j/crypto/Sign.java:
```diff
@@ -181,6 +169,67 @@ public static ECPublicKey recoverFromSignature(int recId, ECDSASignature sig, by
     }
 
+    /**
+     * Given an arbitrary piece of text and an NEO message signature encoded in bytes, returns the public key that
+     * was used to sign it. This can then be compared to the expected public key to determine if the signature was
+     * correct.
+     *
+     * @param message       the encoded message.
+     * @param signatureData the message signature components.
+     * @return the public key used to sign the message.
+     * @throws SignatureException if the public key could not be recovered or if there was a signature format error.
```
Sorry to be so picky, but the docs are a bit confusing to me. I would simplify to something like "Recovers the public key corresponding to the given signature and message."

The first `@param ` says "encoded message" is it a special encoding? if not then I would simplify to "the message".

gsmachado replied:
Agree with @csmuller.
